data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoServiceError.java

```diff
@@ -9,21 +9,46 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/*
+  This class represents an error that returned from the service
+ */
 public class KustoServiceError extends Exception {
-    private ArrayList<Exception> exceptions = null;
+    private ArrayList<Exception> exceptions;
 
-    public KustoServiceError(JSONArray exceptions) throws JSONException {
+    public KustoServiceError(JSONArray jsonExceptions, boolean isOneApi, String message) throws JSONException {
+        this(message);
         this.exceptions = new ArrayList<>();
-        for (int j = 0; j < exceptions.length(); j++) {
-            this.exceptions.add(new Exception(exceptions.getString(j)));
+        if (isOneApi){
+            for (int j = 0; j < jsonExceptions.length(); j++) {
+                this.exceptions.add(new DataWebException(jsonExceptions.getJSONObject(j).toString()));
+            }
+        } else {
+            for (int j = 0; j < jsonExceptions.length(); j++) {
+                this.exceptions.add(new Exception(jsonExceptions.getString(j)));
+            }
         }
-
     }
+
     public KustoServiceError(String message) {
         super(message);
+        this.exceptions = new ArrayList<>();
     }
 
     public List<Exception> getExceptions() {
         return exceptions;
     }
+
+    @Override
+    public String toString() {
+        return exceptions.size() == 0 ? getMessage() : "exceptions\":" + exceptions + "}";
+    }
+
+
+    public TriState isPermanent(){
```

Comment:
I think this is another support for why this class is unnecessary. It doesn't inherit from KustoClientException, so it duplicates this "isPermanent()" concept. Yet if we merge this with DataServiceException, this is included, and we won't have this conceptual duplication of this concept of permanence in separate Kusto exception classes.